modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManager.java

```diff
@@ -685,14 +687,23 @@ public SCIMResponse updateWithPATCH(String existingId, String scimObjectString,
 
                     }
                 } else if (operation.getOperation().equals(SCIMConstants.OperationalConstants.REMOVE)) {
-                    if (newUser == null) {
-                        newUser = (User) PatchOperationUtil.doPatchRemove(operation, oldUser, copyOfOldUser, schema);
-                        copyOfOldUser = (User) CopyUtil.deepCopy(newUser);
-
-                    } else {
-                        newUser = (User) PatchOperationUtil.doPatchRemove(operation, newUser, copyOfOldUser, schema);
-                        copyOfOldUser = (User) CopyUtil.deepCopy(newUser);
+                    try {
+                        if (newUser == null) {
+                            newUser = (User) PatchOperationUtil.doPatchRemove(operation, oldUser, copyOfOldUser,
+                                    schema);
+                            copyOfOldUser = (User) CopyUtil.deepCopy(newUser);
+
+                        } else {
+                            newUser = (User) PatchOperationUtil.doPatchRemove(operation, newUser, copyOfOldUser,
+                                    schema);
+                            copyOfOldUser = (User) CopyUtil.deepCopy(newUser);
+                        }
+                    } catch (BadRequestException e) {
```

Comment:
Can there be any other reason for bad request other than this scenario? If possible it's better to verify a error code or something to uniquely identify this particular scenario.

Reply:
Added a new condition to see if the bad request is generated to invalid path. 
ref: 1598c3ffc439ecee8d577009cbaae3cc662cb67e